python/ray/data/_internal/block_batching/block_batching.py

```diff
@@ -194,9 +194,10 @@ def _prefetch_blocks(
     while sliding_window:
         block_ref = sliding_window.popleft()
         try:
-            sliding_window.append(next(block_ref_iter))
+            next_block = next(block_ref_iter)
+            sliding_window.append(next_block)
             with stats.iter_wait_s.timer() if stats else nullcontext():
-                prefetcher.prefetch_blocks(list(sliding_window))
+                prefetcher.prefetch_blocks([next_block])
```

Comment:
why is this change required instead of triggering the fetch of the entire sliding window?

Reply:
It's a bug previously. There is no need to redundantly prefetching the same blocks (both for wait and actor-based prefetchers).